h2o-core/src/test/java/hex/ModelBuilderTest.java

```diff
@@ -231,6 +234,58 @@ public void testScoreReorderedDomain() {
     }
   }
   
+  @Test
+  public void testToDeleteOK() {
+    testToDelete(false);
+  }
+
+  @Test
+  public void testToDeleteFailing() {
+    testToDelete(true);
+  }
+
+  private void testToDelete(boolean killLocalPOJO) {
+    final boolean multinode = H2O.getCloudSize() > 1;
+    Frame train = null, test = null;
+    Key<?> k = null;
+    try {
+      do {
+        k = Key.make();
+      } while (k.home() && multinode); // prefer the key to be homed elsewhere but make the test work in single-node too
+
+      train = TestFrameCatalog.oneChunkFewRows();
+
+      DummyModelParameters p = new DummyModelParameters();
+      p._makeModel = true;
+      p._train = train._key;
+      p._response_column = train.name(0);
+      
+      DummyModel m = new DummyModelBuilder(p, (Key) k).trainModel().get();
+
+      test = new TestFrameBuilder()
+              .withColNames(train.lastVecName())
+              .withVecTypes(Vec.T_CAT)
+              .withDomain(0, new String[] {"unseen"})
+              .withDataForCol(0, new String[] {"unseen"})
+              .build();
+
+      m.score(test).delete();
+
+      Value modelValue = H2O.STORE.get(m._key);
+      if (modelValue != null && killLocalPOJO) {
+        // kill the local POJO (and with it any state that was not propagated to DKV)
+        modelValue.freePOJO();
+      }
+    } finally {
+      if (train != null)
```

Comment:
why not use scope?

Reply:
Because that can hide issues, explicit key management is clearer when you are working with leaks IMHO.